sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Watch.java

```diff
@@ -803,14 +870,18 @@ public String toString(Growth.TerminationCondition<?, TerminationStateT> termina
     @Nullable private Instant pollWatermark;
     private boolean shouldStop = false;
 
-    GrowthTracker(final Coder<OutputT> outputCoder, GrowthState<OutputT, TerminationStateT> state,
-                  Growth.TerminationCondition<?, TerminationStateT> terminationCondition) {
+    GrowthTracker(
+        final SerializableFunction<OutputT, KeyT> keyFn,
+        final Coder<KeyT> keyCoder,
+        GrowthState<OutputT, KeyT, TerminationStateT> state,
+        Growth.TerminationCondition<?, TerminationStateT> terminationCondition) {
       this.coderFunnel =
           new Funnel<OutputT>() {
             @Override
             public void funnel(OutputT from, PrimitiveSink into) {
               try {
-                outputCoder.encode(from, Funnels.asOutputStream(into));
+                keyCoder.encode(
+                    keyFn.apply(from), Funnels.asOutputStream(into));
```

Comment:
<!--new_thread; commit:1199ebf360c6a28d4e580be6c561bc1218726828; resolved:0-->
Add a comment here ? (this seems to be the crux of the implementation).

Reply:
Done.